src/libraries/Microsoft.PowerFx.LanguageServerProtocol/Handlers/Fx2Nl/BackwardsCompatibleFx2NlLanguageServerOperationHandler.cs

```diff
@@ -0,0 +1,45 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.LanguageServerProtocol.Protocol;
+
+namespace Microsoft.PowerFx.LanguageServerProtocol.Handlers
+{
+    /// <summary>
+    /// A handler that makes newly structured Fx2NL operations backwards compatible with the existing Language Server SDK.
+    /// Sealed class because it is not meant to be inherited and only used for maintaining backwards compatibility.
+    /// </summary>
+    internal sealed class BackwardsCompatibleFx2NlLanguageServerOperationHandler : BaseFx2NlLanguageServerOperationHandler
+    {
+        private readonly INLHandlerFactory _nLHandlerFactory;
+        private NLHandler _nlHandler;
+
+        public BackwardsCompatibleFx2NlLanguageServerOperationHandler(INLHandlerFactory nLHandlerFactory)
+        {
+            _nLHandlerFactory = nLHandlerFactory;
+        }
+
+        protected override async Task<bool> PreHandleFx2NlAsync(LanguageServerOperationContext operationContext, CancellationToken cancellationToken)
+        {
+            // Existing SDK relies on GetNLHandler to get the handler for Nl2Fx operation.
+            // This is a legacy handler and not to be confused with the new handler pattern.
+            _nlHandler = _nLHandlerFactory.GetNLHandler(operationContext.GetScope(_fx2NlRequestParams.TextDocument.Uri), _fx2NlRequestParams) ?? throw new NullReferenceException("No suitable handler found to handle Fx2Nl");

```

Comment:
>_nlHandler [](http://example.com/codeflow?start=12&length=10)

Can we gaurantee that this field is not overwritten by 2 handlers running concurrently?
Could we set it back  to null at line 42, and assert it's null here?